creator/ingest_runs/genomic_data_loader.py

```diff
@@ -0,0 +1,397 @@
+        )
+
+        def get_kf_id(link):
+            """
+            The links given by the DataService yield strings of the form
+            /entity-type/kf_id. This function will return the kf_id.
+            """
+            return link.split('/')[-1]
+
+        dfs = {}
+        endpoints = {
+            GEN_FILES,
+            SEQ_EXP_GEN_FILES,
+            SEQ_EXPS,
+        }
+
+        # Steps 1-3
+        filter_dict = {}
+        for endpoint in endpoints:
+            if endpoint == GEN_FILES:
+                filter_dict = {"harmonized": harmonized}
+            dfs[endpoint] = utils.get_entities(
+                DATASERVICE_URL, endpoint,
+                self.study_id, filter_dict=filter_dict
+            )
+
+        # Step 4
+        # Source genomic files
+        gdf = dfs[GEN_FILES][["kf_id", "external_id"]].rename(
+            columns={
+                "kf_id": "gf_kf_id",
+                "external_id": CONCEPT.GENOMIC_FILE.SOURCE_FILE
+            }
+        )
+        # Sequencing experiments
+        sedf = dfs[SEQ_EXPS][
+            ["kf_id", "external_id", "_links.sequencing_center"]
+        ].rename(
+            columns={
+                "kf_id": CONCEPT.SEQUENCING.TARGET_SERVICE_ID,
+                "external_id": CONCEPT.SEQUENCING.ID,
+            }
+        )
+        SCTID = CONCEPT.SEQUENCING.CENTER.TARGET_SERVICE_ID
+        sedf[SCTID] = sedf["_links.sequencing_center"].apply(
+            lambda link: get_kf_id(link)
+        )
+
+        # Sequencing experiment (source) genomic files
+        sgdf = dfs[SEQ_EXP_GEN_FILES][
+            [
+                "_links.sequencing_experiment",
+                "_links.genomic_file",
+            ]
+        ]
+        sgdf["gf_kf_id"] = sgdf["_links.genomic_file"].apply(
+            lambda link: get_kf_id(link)
+        )
+        sgdf[CONCEPT.SEQUENCING.TARGET_SERVICE_ID] = sgdf[
+            "_links.sequencing_experiment"].apply(
+            lambda link: get_kf_id(link)
+        )
+        # Merge se-gf with se
+        sgdf = pd.merge(sedf, sgdf, on=CONCEPT.SEQUENCING.TARGET_SERVICE_ID)
+
+        # Step 5
+        return pd.merge(gdf, sgdf, on="gf_kf_id")[
+            [
+                CONCEPT.SEQUENCING.TARGET_SERVICE_ID,
+                CONCEPT.SEQUENCING.CENTER.TARGET_SERVICE_ID,
+                CONCEPT.SEQUENCING.ID,
+                CONCEPT.GENOMIC_FILE.SOURCE_FILE,
+            ]
+        ]
+
+    def _load_entities(self, entity_type, df):
+        """
+        Uses the ingest lib's load stage to load the entities of _entity_type_
+        from DataFrame _df_ into the Data Service
+        """
+        self.loader = LoadStage(
+            self.target_api_cfg,
+            DATASERVICE_URL,
+            [entity_type],
+            self.study_id,
+            cache_dir=os.getcwd(),
+        )
+        self.loader.run({entity_type: df})
+        return self.loader.uid_cache
+
+    def _validate_load(self, df):
```

Comment:
We should have a sanity check that will be called after the whole ingest process. Don't know what should go in here yet